### PR TITLE
Fix sync of already persisted entities

### PR DESCRIPTION
**Pull request by sihu**
* if you use assign_attributes on a relation, it is not stored right away. so we have to take care of that manually
* i improve logging (Rails.logger was not logged until now)

### CODE REVIEW

File: app/services/midata_service.rb [hunk 1]
```diff
@@ -7,7 +7,12 @@ class MidataService
   base_uri ENV.fetch('MIDATA_BASE_URL', 'https://pbs.puzzle.ch')
 
   def initialize(user_email = ENV['MIDATA_USER_EMAIL'], user_token = ENV['MIDATA_USER_TOKEN'], _locale = 'de')
-    @auth_params = { 'X-User-Token': user_token, 'X-User-Email': user_email }
+    # Hitobito deprecated user tokens. But this will not be deployed before BuLa
+    @auth_params = if self.class.base_uri.include?('pbs.puzzle.ch')
+                     { Authorization: "Bearer #{ENV['MIDATA_OAUTH_ACCESS_TOKEN']}" }
```
Comment (sihu) [inline app/services/midata_service.rb:12]: if you ask yourself: sadly the current version of MiData does not support User-Tokens. Service-Tokens do not have enough rights for participants of camps. therefore i made this temporary fix. we might have to use such a token with a very long expiration time as well for the sync.

File: app/services/midata_service.rb [hunk 2]
```diff
@@ -32,11 +37,21 @@ def fetch_camp_unit_data_hierarchy(root_url, root: true)
 
     return root_data if children_ids.nil? && !root
 
+    [root ? nil : root_data, children_data(children_urls(children_ids, root_data))].flatten.compact
```
Comment (sihu) [inline app/services/midata_service.rb:40]: @lukaseisenring i guess that was the reason that the root camps always got imported.

File: .env.example [hunk 3]
```diff
@@ -14,6 +14,7 @@ DEVISE_SECRET_KEY=
 MIDATA_BASE_URL=https://pbs.puzzle.ch
 MIDATA_USER_EMAIL=hussein_kohlmann@hitobito.example.com
 MIDATA_USER_TOKEN=Tu7aVJWyLYYMyCnZv2bz
+MIDATA_OAUTH_ACCESS_TOKEN=
```
Comment (sihu) [inline .env.example:17]: further explanations in the spec/services/camp_unit_puller_spec.rb

File: app/services/midata_service.rb (hunk 1 above)
Comment (diegosteiner) [reply app/services/midata_service.rb:12]: Why did it work until now? Or why didn't we notice?

**Review by diegosteiner — APPROVED** [review]

File: app/services/midata_service.rb (hunk 1 above)
Comment (sihu) [reply app/services/midata_service.rb:12]: => pbs.puzzle.ch is already the version where it's not supported whereas https://db.scout.ch still does. as far as i know, they planned to not release before the bula. but we need to make sure!

**Pull request merged**